src/ray/raylet/agent_manager.cc

```diff
@@ -128,15 +128,21 @@ void AgentManager::StartAgent() {
 
     int exit_code = child.Wait();
     timer->cancel();
-    RAY_LOG(WARNING) << "Agent process with id " << agent_id << " exited, return value "
-                     << exit_code << ". ip " << reported_agent_ip_address_ << ". id "
-                     << reported_agent_id_;
+    RAY_LOG(INFO) << "Agent process with id " << agent_id << " exited, exit code "
+                  << exit_code << ". ip " << reported_agent_ip_address_ << ". id "
+                  << reported_agent_id_;
+
     RAY_LOG(ERROR)
         << "The raylet exited immediately because the Ray agent failed. "
            "The raylet fate shares with the agent. This can happen because the "
```

Comment:
> basically, the agent will exit with exit code 0 (normal exit) upon sigterm. If the exit code is 0, raylet performs the graceful shutdown rather than QuickExit (raylet QuickExits when the agent is killed now). I am open to using other exit codes.

Seems we are not checking the exit_code of child?